### ISSUE TITLE
Add 'invalid' to 'ffpe-trap' list for gnu in CESM, once a working gnu version is widely available

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#1974]
I'd like to be able to add `invalid` to the `ffpe-trap` list for gnu. However, this currently causes problems: gfortran's isnan (which is called in cime via the CPRGNU-specific shr_infnan_isnan) causes a floating point exception when called on a signaling NaN. We use `isnan` in various places in CESM, so this would presumably cause problems when running in debug mode.

This appears to be a known gnu bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66462

Once gnu fixes that bug and we can rely on having versions of gcc/gfortran with that bug fixed, we should add `invalid` to the `ffpe-trap` list for gnu.

(This is a revision of #1763 . See that issue for details of things I tried, unsuccessfully, to workaround gnu's current issues.)

**Comment by ekluzek** [comment]
Could this be done for CESM2.0?

**Comment by billsacks** [comment]
@ekluzek no, this relies on a fix to the gnu compiler which I don't think is in place yet (at least not in the versions of gnu that we typically use).
